### PR TITLE
Add new target support for ArrayBuffer and TypedArray

### PR DESCRIPTION
**Pull request by kisbg**
JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-helpers.c [hunk 1]
```diff
@@ -39,12 +41,24 @@ ecma_typedarray_helper_dispatch_construct (const ecma_value_t *arguments_list_p,
   JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
 
   ecma_object_t *prototype_obj_p = ecma_builtin_get (ecma_typedarray_helper_get_prototype_id (typedarray_id));
+
+  if (JERRY_CONTEXT (current_new_target))
+  {
+    prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
+                                                              ecma_typedarray_helper_get_prototype_id (typedarray_id));
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-helpers.c:48]: Save proto_id into a local variable.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-arraybuffer-object.c [hunk 2]
```diff
@@ -141,7 +143,23 @@ ecma_op_create_arraybuffer_object (const ecma_value_t *arguments_list_p, /**< li
...
+                                                                   ECMA_BUILTIN_ID_ARRAYBUFFER_PROTOTYPE);
+
+  if (proto_p == NULL)
+  {
+    return ECMA_VALUE_ERROR;
+  }
+
+  ecma_object_t *array_buffer = ecma_arraybuffer_new_object (length_uint32);
+
+  if (!ECMA_IS_VALUE_ERROR (ecma_make_object_value (array_buffer)))
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-arraybuffer-object.c:156]: This operation cannot fail. So this check is unnecessary.

**Comment by kisbg** [comment]
@rerobika Thank you for the review!

**Review by rerobika — APPROVED** [review]
LGTM

**Pull request merged**